c_sharp_json_diff/JsonDiff.cs

```diff
@@ -1,55 +1,117 @@
+        {
+            string diff = "";
+            try
             {
-                if (firstKeys.Contains(key))
+                diff = JsonDiffBetweenTwo(left, right);
+                if (diff == null)
                 {
-                    matches.Add(key);
+                    return JObject.Parse(@"{}");
                 }
+                JObject diffJsonObject = JObject.Parse(diff);
+                JObject diffReturnObject = CalculateNumericDiff(diffJsonObject, _isAbsoluteDiff);
+                return diffReturnObject;
             }
-            foreach(string key in matches)
+            catch (Exception e)
             {
-                first.TryGetValue(key, out double firstValue);
-                second.TryGetValue(key, out double secondValue);
-                if(firstValue != secondValue)
+                return JObject.Parse(diff + e.StackTrace);
+            }
+        }
+
+        public string NumericDiffAsString(string left, string right)
+        {
+            JObject diff = NumericDiff(left, right);
+            return _isDiffIndented ? diff.ToString(Formatting.Indented) : diff.ToString();
+
+        }
+
+        private JObject CalculateNumericDiff(JObject jObject, bool isAbsoluteDiff)
+        {
+            foreach (KeyValuePair<string, JToken?> prop in jObject)
+            {
+                if (prop.Key == "_t")
+                {
+                    continue;
+                }
+                if (prop.Value != null)
                 {
-                    Diff item = new Diff();
-                    item.jsonProperty = key;
-                    item.previous = firstValue;
-                    item.current = secondValue;
-                    item.difference = Math.Abs(firstValue - secondValue);
-                    result.Diffs.Add(item);
+                    if (prop.Value is JArray)
+                    {
+                        JArray valuesAsArray = (JArray)prop.Value;
+                        if (valuesAsArray.Count >= 2)
+                        {
+                            if (valuesAsArray[0] != null && valuesAsArray[1] != null)
+                            {
+                                string leftValueAsString = valuesAsArray[0].ToString();
+                                string rightValueAsString = valuesAsArray[1].ToString();
+                                if (int.TryParse(leftValueAsString, out int leftValueAsInt) && int.TryParse(rightValueAsString, out int rightValueAsInt))
+                                {
+                                    int diffValue = isAbsoluteDiff ? Math.Abs(leftValueAsInt - rightValueAsInt) : leftValueAsInt - rightValueAsInt;
+                                    Dictionary<string, int> map = CreateDictionary<string, int>("left", "right", "diff", leftValueAsInt, rightValueAsInt, diffValue);
+                                    valuesAsArray.RemoveAll();
+                                    valuesAsArray.Add(JToken.FromObject(map));
```

Comment:
As I mentioned in the test code, would it not make more sense to replace the array with the dictionary? Instead of always creating a list of length 1?

Reply:
Yeah that makes more sense, I didn't test adding dictionary directly back to the Json object instead of the array. It seems to work fine when I add dictionary directly